staging/src/k8s.io/cloud-provider/cloud.go

```diff
@@ -294,4 +294,15 @@ type InstanceMetadata struct {
 	// NodeAddress contains information for the instance's address.
 	// The node addresses returned here will be set on the node's status.addresses field.
 	NodeAddresses []v1.NodeAddress
+
+	// Zone is the zone that the instance is in.
+	// The value set here is applied as the following labels on the node:
+	//   * topology.kubernetes.io/zone=<zone>
+	//   * failure-domain.beta.kubernetes.io/zone=<zone> (DEPRECATED)
+	Zone string
```

Comment:
I'm fine with adding those (that sounds like a good direction to me), but given that all in-tree providers already support "GetInstanceMetadata", this basically will break supportability of it.

Can you at least add appropriate TODOs for all cloud providers in the InstancesV2 method that those two will have to be propagated before enabling the inferface?

Reply:
> I'm fine with adding those (that sounds like a good direction to me), but given that all in-tree providers already support "GetInstanceMetadata", this basically will break supportability of it.

So in https://github.com/kubernetes/kubernetes/pull/93582 I actually removed all the implementations of `InstanceMetadataByProviderID` since that code path isn't actually run anywhere. 

> Can you at least add appropriate TODOs for all cloud providers in the InstancesV2 method that those two will have to be propagated before enabling the inferface?

Also added TODOs comments in https://github.com/kubernetes/kubernetes/pull/93582 for providerse to implement InstanceV2 when out-of-tree.

Reply:
As you correctly point out, this API is still marked as experimental. Do we have a graduation plan? It seems to suggest (but does not explicitly state) we should be graduating in 1.20